front/src/reducers/osrdconf/stdcmConf/index.ts

```diff
@@ -94,10 +94,15 @@ export const stdcmConfSlice = createSlice({
     },
     updateStdcmPathStep(
       state: Draft<OsrdStdcmConfState>,
-      action: PayloadAction<ArrayElement<OsrdStdcmConfState['stdcmPathSteps']>>
+      action: PayloadAction<{
+        id: string;
+        updates: Partial<ArrayElement<OsrdStdcmConfState['stdcmPathSteps']>>;
+      }>
     ) {
       const newPathSteps = state.stdcmPathSteps.map((pathStep) =>
-        pathStep.id === action.payload.id ? action.payload : pathStep
+        pathStep.id === action.payload.id
+          ? ({ ...pathStep, ...action.payload.updates } as StdcmPathStep)
```

Comment:
Nit: since we're spreading `pathStep`, do we really need `as StdcmPathStep`?

Reply:
Yes we need it, because of the type `isVia` (the props are not the same depending on its value, so TS is not very happy about it) :/